src/metldata/builtin_transformations/sum_operation/data_transform.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2021 - 2024 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Data transformation logic for sum operation transformation."""
+
+from schemapack.spec.datapack import DataPack
+
+from metldata.builtin_transformations.common.data_transform import transform_data
+from metldata.builtin_transformations.sum_operation.instruction import (
+    SumOperationInstruction,
+)
+
+
+def sum_content(
+    *, data: DataPack, instructions_by_class: dict[str, list[SumOperationInstruction]]
+) -> DataPack:
+    """Apply all transformation instructions."""
+    return transform_data(
+        data=data,
+        instructions_by_class=instructions_by_class,
+        calculate_value=sum_values,
+    )
+
+
+def sum_values(values_to_sum: list[int | float | bool]) -> int | float:
+    """Sum up the values given in a list."""
+    return sum(val for val in values_to_sum if isinstance(val, int | float | bool))
```

Comment:
Could use a type alias for `int | float | bool` here and filter might express intent a bit better:

```suggestion
def sum_values(values_to_sum: list[int | float | bool]) -> int | float:
    """Sum up the values given in a list."""
    return sum(filter(values_to_sum, lambda val: isinstance(val, int | float | bool)))
```

Reply:
On a second thought, we don't need the `isinstance` part at all. if the data type does not allow sum operation, the assumption fails. If the transformed datapack contains a value from a wrong data type, datapack's validation against its schemapack fails. A type check here seems redundant.